docs/authentication/stytch/permit-integration.mdx

```diff
@@ -0,0 +1,349 @@
+      });
+    }
+  }, [router]);
+
+  return <div>Loading...</div>;
+}
+```
+
+### Step 5: Grant Access to Protected Frontend Content
+
+Upon successful authentication, our frontend JavaScript SDK will automatically populate the session and user objects. Use the
+useStytchSession and useStytchUser hooks to listen for changes:
+
+```javascript
+import { useStytchUser, useStytchSession } from "@stytch/nextjs";
+import { useEffect } from "react";
+import { useRouter } from "next/router";
+
+export default function ProfilePage() {
+  const { user, isInitialized } = useStytchUser();
+  const router = useRouter();
+
+  useEffect(() => {
+    if (isInitialized && !user) {
+      router.replace("/");
+    }
+  }, [user, isInitialized, router]);
+
+  return <div>Profile Page Content</div>;
+}
+```
+
+### Step 6: Grant Access to Protected Server-side Routes
+
+Load our backend Node SDK, retrieve the Stytch session_token or session_jwt from the request cookies, and then call the
+`sessions.authenticate` or `sessions.authenticateJwt` methods:
+
+```javascript
+import stytch from "stytch";
+
+let client;
+
+const loadStytch = () => {
+  if (!client) {
+    client = new stytch.Client({
+      project_id: process.env.STYTCH_PROJECT_ID || "",
+      secret: process.env.STYTCH_SECRET || "",
+      env: process.env.STYTCH_PROJECT_ENV === "live" ? stytch.envs.live : stytch.envs.test,
+    });
+  }
+  return client;
+};
+
+export async function getServerSideProps({ req }) {
+  const redirectRes = {
+    redirect: {
+      destination: "/",
+      permanent: false,
+    },
+  };
+
+  const sessionJWT = req.cookies["stytch_session_jwt"];
+
+  if (!sessionJWT) {
+    return redirectRes;
+  }
+
+  const stytchClient = loadStytch();
+
+  try {
+    await stytchClient.sessions.authenticateJwt(sessionJWT);
+    return { props: {} };
+  } catch (e) {
+    return redirectRes;
+  }
+}
+```
+
+As you saw in the above step, we are fetching a `sessionJWT` directly form the cookies. As we are also able to authenticate the session
+with the `.authenticateJwt()` function, we therefore know the `user id` will also be available to us, so we can proceed to sync the user
+inside of Permit. This is part of the **handoff point**.
+
+Before we move to utilize the syncUser function, we need to do a basic Permit setup in the UI.
+
+## Permit Pre-sync setup guide
+
+For the initial setup guide with Permit, please follow the [Quickstart guide](/quickstart). It will give you an outline how to
+create a basic `workspace` with permit, create roles, resources, and assign permissions to the roles.
+
+Below is an example of the policy we will be working with. The only think we are missing is syncing a user into Permit, and assigning
+them the appropriate role.
+
+![Basic Policy](/ui-videos/authentication/stytch/account-owner-role.png)
+
+### Importing and initializing the Permit object
+
+Inside of your backend file, we need to import the Permit package and initialize the Permit instance object.
```

Comment:
Where is this backend file? Are we following a practical example or using general Next.js project?